src/SuperDumpService/Controllers/HomeController.cs

```diff
@@ -343,6 +351,27 @@ public class HomeController : Controller {
 			var id = DumpIdentifier.Create(bundleId, dumpId);
 			superDumpRepo.RerunAnalysis(id);
 			return View(new ReportViewModel(id));
+		}
+
+		[Authorize(Policy = LdapCookieAuthenticationExtension.UserPolicy)]
+		[HttpPost]
+		public IActionResult ExtendRetentionTime(string bundleId, string dumpId) {
+			var bundleInfo = superDumpRepo.GetBundle(bundleId);
+			if (bundleInfo == null) {
+				logger.LogNotFound("ExtendRetentionTime: Bundle not found", HttpContext, "BundleId", bundleId);
+				return View(null);
+			}
+			var id = DumpIdentifier.Create(bundleId, dumpId);
+			var dumpInfo = superDumpRepo.GetDump(id);
+			if (dumpInfo == null) {
+				logger.LogNotFound("Report: Dump not found", HttpContext, "Id", id.ToString());
+				return View(null);
+			}
+			logger.LogDumpAccess("ExtendRetentionTime", HttpContext, bundleInfo, dumpId);
+
+			superDumpRepo.ExtendDumpRetentionTime(id, $"The Retention Time was extended by user {HttpContext.User.Identity.Name} for 30 days", TimeSpan.FromDays(30));
```

Comment:
Do not add 30 days every time. Set it to Now+30 days. We don't want users to click 100x and thus set retention to 3000 days.